homeassistant/components/zha/__init__.py

```diff
@@ -91,13 +92,13 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     Will automatically load components to support devices found on the network.
     """
 
-    # Strip whitespace around `socket://` URIs, this is no longer accepted by zigpy
-    # This will be removed in 2023.7.0
+    # Remove brackets around IP addresses, this no longer works in CPython 3.11.4
+    # This will be removed in 2023.11.0
     path = config_entry.data[CONF_DEVICE][CONF_DEVICE_PATH]
     data = copy.deepcopy(dict(config_entry.data))
 
-    if path.startswith("socket://") and path != path.strip():
-        data[CONF_DEVICE][CONF_DEVICE_PATH] = path.strip()
+    if re.match(r"^(socket|tcp)://\[\d+\.\d+\.\d+\.\d+\]:\d+$", path):
```

Comment:
Can the path have brackets and a whitespace? Or is it assumed that any whitespace was removed previously?

Reply:
I'm not sure how either get in so it's possible but I believe the whitespace stripping code was present for about four months before this, which should be ample time to migrate existing entries.

Reply:
I'm fine with this approach. 
Alternatively, what if we modify the regex with `\s*$` at the end and if matched, then replace brackets and strip().
Just for those who upgrade once every 4 month :)